public/hooks/use_save_chat.test.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { renderHook } from '@testing-library/react-hooks';
+import { NOTEBOOK_API } from '../../common/constants/llm';
+
+import { useSaveChat } from './use_save_chat';
+import * as chatStateHookExports from './use_chat_state';
+import * as coreHookExports from '../contexts/core_context';
+import { httpServiceMock } from '../../../../src/core/public/mocks';
+import * as chatContextHookExports from '../contexts/chat_context';
+
+describe('useSaveChat hook', () => {
+  const chatName = 'foo';
+  const httpMock = httpServiceMock.createStartContract();
+  const chatStateDispatchMock = jest.fn();
+
+  const chatContextMock = {
+    rootAgentId: 'root_agent_id_mock',
+    selectedTabId: 'chat',
+    setSessionId: jest.fn(),
+    setTitle: jest.fn(),
+    currentAccount: { username: 'admin' },
+  };
+
+  beforeEach(() => {
+    jest.spyOn(coreHookExports, 'useCore').mockReturnValue({
+      services: {
+        http: httpMock,
+      },
+    });
+    jest.spyOn(chatContextHookExports, 'useChatContext').mockReturnValue(chatContextMock);
+
+    jest.spyOn(chatStateHookExports, 'useChatState').mockReturnValue({
+      chatState: { messages: [], interactions: [], llmResponding: false },
+      chatStateDispatch: chatStateDispatchMock,
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should throw error if createNotebook function does not return id', async () => {
+    httpMock.post.mockResolvedValueOnce(null);
+    const { result } = renderHook(() => useSaveChat());
+    const saveChat = result.current.saveChat;
+    try {
+      await saveChat(chatName);
+    } catch (e) {
```

Comment:
The test case will be succeed if `saveChat` doesn't throw any error. It's not an expected behavior.
Prefer move expect outside of catch scope. Sample code:
```tsx
let saveChatError;
try {
    await saveChat(chatName);
} catch (e) {
    saveChatError = e;
}
expect(httpMock.post).toHaveBeenCalledWith(NOTEBOOK_API.CREATE_NOTEBOOK, {
  body: JSON.stringify({ name: chatName }),
});
expect(saveChatError?.message).toBe('create notebook error');
```
Or may be we can change to like [this blog](https://blog.codeleak.pl/2021/11/testing-exceptions-in-javascript.html).

Reply:
Agree, I prefer toThrow assertions